build/configs/rtl8721csm/rtl8721csm_download.sh

```diff
@@ -243,8 +244,16 @@ function get_partition_sizes()
 cp -p ${BIN_PATH}/km0_boot_all.bin ${IMG_TOOL_PATH}/km0_boot_all.bin
 cp -p ${BIN_PATH}/km4_boot_all.bin ${IMG_TOOL_PATH}/km4_boot_all.bin
 cp -p ${BIN_PATH}/km0_km4_image2.bin ${IMG_TOOL_PATH}/km0_km4_image2.bin
-if [[ "${CONFIG_APP_BINARY_SEPARATION}" == "y" ]];then
+if test -f "${SMARTFS_BIN_PATH}"; then
 	cp -p ${BIN_PATH}/rtl8721csm_smartfs.bin ${IMG_TOOL_PATH}/rtl8721csm_smartfs.bin
+else
+cat <<EOF
+Warning!! Smartfs Binary does not exist.
```

Comment:
As you did before, smartfs is not mandatory to flash. Could you confirm this is working with `make download all` without smartfs image?

Reply:
I think this logic should be in the sanity check function with smartfs selected.

Reply:
You are correct it will pose problem to make download ALL.
I will modify this